Python/import.c

```diff
@@ -2308,13 +2308,11 @@ PyImport_ExtendInittab(struct _inittab *newtab)
     _PyMem_SetDefaultAllocator(PYMEM_DOMAIN_RAW, &old_alloc);
 
     /* Allocate new memory for the combined table */
-    if ((i + n + 1) <= PY_SSIZE_T_MAX / (Py_ssize_t)sizeof(struct _inittab)) {
+    p = NULL;
+    if (i + n <= SIZE_MAX / sizeof(struct _inittab) - 1) {
```

Comment:
```c
if (i <= PY_SSIZE_T_MAX / sizeof(struct _inittab) - 1 - n) {
```

Reply:
`i + n` should be fine because both arrays are already stored in memory.

Reply:
I did have the same concern. Then maybe remove this check at all? The sizes of existing archives are `(i + 1) * sizeof(struct _inittab)` and `(n + 1) * sizeof(struct _inittab)`.

Reply:
i+n+1 is fine because we already talking about builtin modules. I expect the list to be shorter than 100. So way smaller than 2^31-1 (and 2^63-1) :-)